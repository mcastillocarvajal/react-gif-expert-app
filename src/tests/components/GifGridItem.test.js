import React from 'react'
import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"


describe('Test for <GifGridItem />', () => {

    const title = 'title';
    const url = 'https://localhost/algo.png'
    const wrapper = shallow( <GifGridItem 
        title={ title }
        url={ url }
    /> )
    
    test('should show component correctly', () => {

        expect( wrapper ).toMatchSnapshot();
    })

    test('should have a <p> with a text', () => {

        const p = wrapper.find('p')

        expect( p.text().trim() ).toBe( title )
    })

    test('should have a <img> with the props url and alt', () => {

        const img = wrapper.find('img')

        expect( img.prop('src') ).toBe( url )
        expect( img.prop('alt') ).toBe( title )
    })

    test('should have the classname with animation', () => {
        
        const div = wrapper.find('div')
        const className = div.prop('className')

        expect( className.includes( 'animate__fadeIn' ) ).toBe( true )
    })
    
    
    
})
