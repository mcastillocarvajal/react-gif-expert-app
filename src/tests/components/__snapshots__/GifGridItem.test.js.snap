// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for <GifGridItem /> should show component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="title"
    src="https://localhost/algo.png"
  />
  <p>
    title
  </p>
</div>
`;
